portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,144 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
- */
+const COMMENT_SIZE_LIMIT = 100;
+
 function getRandomFact() {
-  const greetings =
-      ['My ex podmate stole my go link', 'I was the shortest person in my junior high', 'I know 3 languages', 'C++ is my favorite programming language'];
+  const facts =
+    ["My ex podmate stole my go link", "I was the shortest person in my junior high", "I know 3 languages", "C++ is my favorite programming language"];
+
+  const fact = facts[Math.floor(Math.random() * facts.length)];
+
+  const factContainer = document.getElementById('fact-container');
+  factContainer.innerText = fact;
+}
+
+const createCommentElement = commentPayload => {
+  console.log(commentPayload);
+  const commentElement = document.createElement("li");
+  const user = document.createElement("h4");
+  const commentWrapper = document.createElement("div");
+  const comment = document.createElement("p");
+  const like = document.createElement("i");
+  const likeIcon = document.createElement("span");
+  const trash = document.createElement("i");
+  const trashIcon = document.createElement("span");
+  const numOfLikes = document.createElement("p");
+  const bottomWrapper = document.createElement("div");
+
+  const commentText = commentPayload.comment;
+  const id = String(commentPayload.id);
+
+  if (commentPayload.size <= COMMENT_SIZE_LIMIT) {
+    comment.innerHTML = commentText;
+    commentWrapper.appendChild(comment);
+  } else {
+    const fullComment = document.createElement("p");
+    const more = document.createElement("i");
+    const moreIcon = document.createElement("span");
+
+    moreIcon.classList.add("glyphicon", "glyphicon-triangle-bottom");
+
+    const shortenedComment = commentText.slice(0, COMMENT_SIZE_LIMIT);
+
+    fullComment.innerHTML = commentText;
+
+    comment.innerHTML = shortenedComment.concat("...");
+    comment.id = id.concat("-visible");
+
+    fullComment.classList.add("hidden-element");
+    fullComment.id = id.concat("-hidden");
+
+    more.appendChild(moreIcon);
+    more.id = id.concat("-btn");
+    more.onclick = () => showFullText(id);
+
+    commentWrapper.appendChild(fullComment);
+    commentWrapper.appendChild(comment);
+    bottomWrapper.appendChild(more);
+  }
+
+  user.innerHTML = "@ ".concat(commentPayload.user);
+
+  likeIcon.classList.add("glyphicon", "glyphicon-thumbs-up");
+  trashIcon.classList.add("glyphicon", "glyphicon-trash");
+
+  trash.appendChild(trashIcon);
+  trash.onclick = () => removeComment(id);
```

Comment:
Since there's no user authentication, I assume anyone can delete anyone else's comments? 

Ideally, only you as an admin or the person who wrote the comment would be able to do that. Could be an interesting addition to consider for the libraries walkthrough week.

Reply:
Yes, however I do not plan to use the authentication library